Feature/#6 api call

reemrev21 opened the pull request:
> ## Summary
- react query + axios 를 이용한 api call
- react query 를 이용한 상태관리

> ## Describe your changes
- search / info 분리해둠 
  - search input box 는 추후 지도 마커로 정해두는게 나을 것 같습니다
  - 하면서 느낀건데 아무것도 검색 안했을 때 초기 화면도 생각하는게 좋을듯 합니다
- react-query 랑 axios 모듈화 좀 더 깔끔하게 해야할듯..🤔

> ## Issue number and link
#6

> ## Merge 진행 또는 Pull 받는 팀원들이 주의해야 할 사항
npm install 
.env.local 설정

reemrev21 committed:
[feature] #6 install react-query

reemrev21 committed:
[feature] #6 setting react-query & remove useless file

reemrev21 committed:
[feature] #6 success api call
- install axios
- react-query 를 이용한 search 상태 관리
- react-query + axios 를 이용한 api call
- 하지만 이 부분 코드 퀄리티 더 높힐 필요성 있음

alsrlqor1007 reviewed (approved):
생각해보니 검색 행위나 절차의 유무도 다시 한번 함께 고려보는 것도 좋을 것 같아요!

reemrev21 committed:
[fix] #6 fix bug
- missing queryFn 에러 수정
  - 상태 관리 시 비동기로 initialState 정해줘야함
- fetch 랑 react-query 분리
  - 파일로 어떻게 분리 할지 고민해봐야함..

reemrev21 committed:
[chore] #6 add map img

byungmin12 commented on client/components/search/Search.tsx:
```diff
@@ -0,0 +1,27 @@
+  const [search, setSearch] = useState("")
+
+  const handleSearch = () => {
+    queryClient.setQueryData("search", search)
```
setQueryData 와 useQuery("search") 이 부분을 다른 상태관리 처럼 재사용이 가능할 것 같이요
예를들어
```js
const useSetValue = (key:string,) => {
    return (value:string)=>queryClient.setQueryData(key,value)
}

const useGetValue = (key:string) => {
    return useQuery(key)
}

const useCustomState = (key:string) => {
    return [ useQuery(key),  (value:string)=>queryClient.setQueryData(key,value)]
}
// 사용법은 정확하지 않을 수 있어요 ... react-query 사용법을 잘 몰라서 하하
```
대충 이런식으로 사용하면 recoil이나 useState처럼 사용이 가능할 것 같아요

reemrev21 replied:
useQueryClient 자체가 hook 이라서 모듈화 하는데 어려움이 크네요.. 혹시 이 부분은 추후에 변경해도 될까요? 🥲

byungmin12 commented on client/api/info.ts:
```diff
@@ -0,0 +1,15 @@
+import { useQuery } from "react-query";
+import api from "./api";
+
+export const fetchHanRiverInfo = async (search: string | undefined) => {
```
아까 말했던 것과 같이 fetchHanRiverInfo 이 부분을 다른 fetch요청에서 재사용할 수 있게 변경할 수 있을 것 같아요 

JDA에서 사용한 것과 비슷하게 

```js
export const Fetcher = function <Type>(props: FecherGetProps) {
  return axios({
    url: `${props.url}`,
    method: "get", // JDA에서는 SWR만 사용하여 get으로 정해두었지만 React-query를 다른 것도 지원하기 떄문에 props로 받아도 될 것 같아요 
    params: props.params
  }).then((res) => res.data as Type);
};

```

이런식으로 fetch를 작성하면 hanriverInfo뿐 아니라 추후에 추가 될 모든 api요청들에서도 함수를 재사용할 수 있을 것 같아요 !

reemrev21 replied:
적용하였습니다 !

reemrev21 committed:
[refactor] #6 코드리뷰 바탕으로 수정
- Fetcher
- next config js 수정
